### PR TITLE
Add support of rcar spider ca55 board

### PR DESCRIPTION
**Pull request by xakep-amatop**

### CODE REVIEW

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: soc/renesas/rcar/rcar_gen4/Kconfig [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
 	select CLOCK_CONTROL_RCAR_CPG_MSSR if CLOCK_CONTROL
 	select ARM_ARCH_TIMER
+
+config SOC_SERIES_RCAR_GEN4_A55
+	bool
+	select ARM64
+	select CPU_CORTEX_A55
+	select CLOCK_CONTROL_RCAR_CPG_MSSR if CLOCK_CONTROL
+	select ARM_ARCH_TIMER
+
```
[the remark this answers is not in the document]
Comment (nordicjm) [reply soc/renesas/rcar/rcar_gen4/Kconfig:18]: That needs fixing

File: drivers/pinctrl/renesas/rcar/CMakeLists.txt [hunk 2]
```diff
@@ -8,4 +8,7 @@ if (CONFIG_SOC_R8A77951_R7 OR CONFIG_SOC_R8A77951_A57)
 endif()
 
 zephyr_library_sources_ifdef(CONFIG_SOC_R8A77961 pfc_r8a77961.c)
-zephyr_library_sources_ifdef(CONFIG_SOC_R8A779F0 pfc_r8a779f0.c)
+
```
[the remark this answers is not in the document]
Comment (xakep-amatop) [reply drivers/pinctrl/renesas/rcar/CMakeLists.txt]: done

File: soc/renesas/rcar/rcar_gen4/Kconfig (hunk 1 above)
[the remark this answers is not in the document]
Comment (xakep-amatop) [reply soc/renesas/rcar/rcar_gen4/Kconfig]: changed

File: boards/renesas/rcar_spider_s4/Kconfig.rcar_spider_s4 [hunk 3]
```diff
@@ -2,4 +2,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_RCAR_SPIDER_S4
-	select SOC_R8A779F0
+	select SOC_R8A779F0_R52 if BOARD_RCAR_SPIDER_S4_R8A779F0_R52
+	select SOC_R8A779F0_A55 if BOARD_RCAR_SPIDER_S4_R8A779F0_A55
+
+config BUILD_OUTPUT_BIN
+	default y if BOARD_RCAR_SPIDER_S4_R8A779F0_A55
```
Comment (nordicjm) [inline boards/renesas/rcar_spider_s4/Kconfig.rcar_spider_s4:9]: `Kconfig`

**Review by nordicjm — CHANGES_REQUESTED** [review]
Minor change then looks good

**Review by firscity — DISMISSED** [review]

**Review by lorc — DISMISSED** [review]

File: boards/renesas/rcar_spider_s4/Kconfig.rcar_spider_s4 (hunk 3 above)
Comment (xakep-amatop) [reply boards/renesas/rcar_spider_s4/Kconfig.rcar_spider_s4]: moved to `Kconfig.defconfig`

**Review by nordicjm — DISMISSED** [review]

**Review by firscity — DISMISSED** [review]

**Review by aaillet — DISMISSED** [review]
LGTM, thanks for this work

**Comment by xakep-amatop** [comment]
Can these changes be merged?

**Comment by xakep-amatop** [comment]
Rebased changes and deleted changes from `sbs_charger` and `fuel_gauge` tests after fixing the issue (https://github.com/zephyrproject-rtos/zephyr/issues/71870) with tests filtering.

**Review by nordicjm — APPROVED** [review]

**Review by firscity — APPROVED** [review]

**Review by aaillet — APPROVED** [review]

**Pull request merged**